youtube_dl/extractor/rte.py

```diff
@@ -60,3 +56,71 @@ def _real_extract(self, url):
             'thumbnail': thumbnail,
             'duration': duration,
         }
+
+
+
+class RteRadioIE(InfoExtractor):
+    IE_NAME = 'rte.ie'
+    IE_DESC = 'Raidió Teilifís Éireann Radio'
+    _VALID_URL = r'''(?x)
+        https?://(?:www\.)?rte\.ie/radio/utils/radioplayer/rteradioweb\.html
+        [#]!rii=(?:[0-9]+)(?:%3A|:)  # Channel ID
+        (?P<id>[0-9]+)(?:%3A|:)      # Item ID
+        (?:[0-9]+)(?:%3A|:)          # Playable item (ie program) ID
+        (?:[0-9]{2}(%2D|-)[0-9]{2}(%2D|-)[0-9]{4})(?:%3A|:) # Start date
```

Comment:
Everything after `id` may be skipped and not matched since http://www.rte.ie/radio/utils/radioplayer/rteradioweb.html#!rii=16:10507902 works just fine.